#Fig. 1
##(A)
**Subsection of a 245 orthologus multi-gene concatenated phylogeny (85 taxa and 71,862 amino acid sites) demonstrating the branching position of Hyphochytrium within the stramenopiles calculated using Maximum Likelihood methods and summary of genome stats.** See Data S1 for full tree file. Dots on the phylogeny indicate 100% bootstrap support, values below 50% have been removed. Cartoons of cells indicate change in flagellum anatomy over the tree. Key indicates generalized lifestyle of the taxa sampled.
##(B)
Table of genome statistics for a range of different stramenopiles.

#Fig. 2
**Comparison of secreted proteome and putative carbohydrate active proteins across the pseudofungi.** A comparative distribution of the proportion of Oomycete proteomes and predicted secretomes which have carbohydrate digesting activity. The schematic phylogeny at the top indicates the relationship between different oomycete species with the dominant lifestyle of each species indicated by text colour. Specifically, green text (Phytophthora species) indicates hemibiotrophy, blue (Hyaloperonospora and Albugo) indicates obligate biotrophy, teal (Pythium) indicates nectrophy, orange (Saprolegnia) indicates both saprotrophy and necrotrophy, finally, black indicates unknown (Hyphochytrium) or non-oomycete (Ectocarpus and Thalassiosira). The first heat map in white/purple indicates the proportion of proteome of each organism (organised columnwise) which was identified as belonging to a particular CAZY (www.cazy.org) database category using BLASTP with an expectation of 1e-5. The number listed is the proportion and the colouring relates to magnitude of the listed number (as shown by scale bar). The second heatmap in blue/yellow is largely the same but instead indicates the proportion of the secretome (predicted via a custom pipeline https://github.com/fmaguire/predict_secretome/tree/refactor) that is identified as belonging to each of these CAZY categories. Finally, the bottom barplot shows the proportion of the secretome for each organism which is predicted to be secreted by the above mentioned pipeline.Analysis includes an outgroup of photosynthetic stramenopiles.

#Fig. 3

#Fig. 4
**Comparative genomic analysis of flagellum proteome evolution.**
##(A)
Heat map showing sequence identity profiles for flagella proteins with putative homologues present across a wide diversity of eukaryotes. The wider analysis uses a seed dataset of 592 proteins (see Table S6 for full dataset). The proteins show proteome loss or conservation patterns consistent with retention or loss of proteins that function in the anterior-tinselated flagellum. The heat map identifies 29 proteins present in the oomycetes but absent in H. catenoides, suggesting that this gene had been lost at the same proximate point to that where the posterior flagellum was also lost. The analysis also shows 12 proteins (marked*) identified as posterior flagellum specific in C. bullosa that are retained in H. catenoides. Three C. bullosa anterior flagellum specific proteins are also retained in H. catenoides. The putative radial spoke proteome also shows numerous losses similar to Homo sapiens (**), this includes the loss of RSP7 (***).
##(B)
Shows a cartoon of the radial spoke protein complex identified in Chlamydomonas (88) with each shape number referring to the RPS number. Black shapes illustrate conserved proteins of the spoke complex, grey are non-conserved proteins, while the white complex refers to RPS7 which although absent in Homo sapiens has been lost separately when the posterior flagellum was also lost. Collectively, this suggests the Hyphochytrium anterior flagellum has a highly reduced flagellum spoke complex.

#Fig. 5
**Photoreceptors putatively encoded on the Hyphochytrium genome.**
##(A)
Distribution of known photoreceptor families across a subset of stramenopile genomes
##(B)
Subsection of the Rhodopsin alignment showing the conserved Schiff base.
##(C)
Phylogeny of the rhodopsin gene family showing phylogenetic position of the hyphochytrium putative rhodopsins.

#Fig. 6
**Comparative genomic analysis of gene families that function in polarised filamentous growth in the fungi.**
##(A)
Cartoon outlining how proteins and complexes involved in polarised growth in Saccharomyces cerevisiae: vesicles are delivered from the Golgi (Fig. 5A-i) along cytoskeleton tracks to predetermined sites on the plasma membrane. Cdc42p is activated by Cdc24p (Fig. 5A-ii) promoting (97) assembly of the polarisome complex (Fig. 5A-iii) resulting in the formin Bni1p radiating actin cables (138, 139). Msb3p and Msb4p interact with Spa2 in the polarisome (Fig. 5A-iv) which is thought to recruit Cdc42 from the cytosol at the site of tip growth (140). Post-Golgi secretory vesicles are transported along the actin cables using a type V myosin (myo2) motor protein (141, 142) (Fig. 5A-v), to dock with the exocyst complex in a process dependent on Sec4 and its GEF Sec2 (143, 144)(Fig. 5A-vi) and so the vesicle is guided to its target site on the plasma membrane (145). Cdc42p along with Rho1  is required for localisation of Sec3p, which acts as a spatial marker for the formation of the exocyst (Fig. 5A-vii) while Rho3 p and Cdc42p mediate docking of the vesicle (Fig. 5A-viii).  Cdc42p plays a key role in regulating these processes in yeast but in Pezizomycotina and Basidiomycete fungi equivalent functions are also performed by Rac1p (146, 147).
##(B)
The domain architecture of the 17 proteins associated with polarised growth in fungi. (C)  The taxon distribution of putative homologues of polarised growth proteins across a representative set of taxa including the Pseudofungi. ‘P’ indicates a putative paralogue relationship as identified using phylogenetic analysis.

#Fig. 7
**Comparative genomic analysis of gene families that function in cell wall synthesis.**
##(A)
Micrographs showing the Wheat Germ Agglutinin fluorescent staining of a chitin cell wall on Hyphochytrium developmental structures.
##(B)
The domain architecture of eight proteins that function in cell wall synthesis.
##(C)
The taxon distribution of putative gene families associated with cell wall synthesis across a representative set of taxa including the Pseudofungi. 
